migrations/db/migrations/20230607094822_add_auth_schema_protections.sql

```diff
@@ -0,0 +1,84 @@
+-- migrate:up
+
+create sequence if not exists
+  auth.changes_seq increment by 1;
+
+comment on sequence auth.changes_seq is 'Tracks the number of changes of the auth schema.';
+
+grant all on sequence
+  auth.changes_seq
+  to supabase_auth_admin
+  with grant option;
+
+grant usage, select on sequence
+  auth.changes_seq
+  to supabase_admin;
+  -- supabase_admin can advance the sequence only
+
+create sequence if not exists
+  auth.lock_seq increment by -1;
+  -- starting off with protection disabled
+
+comment on sequence auth.lock_seq is 'Used to protect against accidental schema changes in the auth schema.';
+
+grant all on sequence
+  auth.lock_seq
+  to supabase_admin, supabase_auth_admin
+  with grant option;
+
+create or replace function
+  auth.enable_schema_ddl_protection()
+  returns void
+  language sql
+  as $$
+    alter sequence if exists auth.lock_seq increment by 1 restart with 1;
+  $$;
+
+grant all on function
+  auth.enable_schema_ddl_protection()
+  to supabase_admin, supabase_auth_admin
+  with grant option;
+
+create or replace function
+  auth.disable_schema_ddl_protection()
+  returns void
+  language sql
+  as $$
+    alter sequence if exists auth.lock_seq increment by -1 restart with -1;
+  $$;
+
+grant all on function
+  auth.disable_schema_ddl_protection()
+  to supabase_admin, supabase_auth_admin
+  with grant option;
+
+create or replace function
+  auth.schema_ddl_protection()
+  returns event_trigger
+  security definer
+  language plpgsql
+  as $$
+    declare
+      cmd record;
+    begin
+      for cmd in select * from pg_event_trigger_ddl_commands()
+      loop
+        if cmd.schema_name = 'auth' then
+          if nextval('auth.lock_seq') > 0 then
+            raise exception 'auth schema is protected from unintended changes. To unlock run SELECT auth.disable_schema_ddl_protection(); ';
+          end if;
+
+          perform nextval('auth.changes_seq');
+        end if;
+      end loop;
+    end;
+  $$;
+
+drop event trigger if exists protect_auth_schema_ddl cascade;
+
+create event trigger
+  protect_auth_schema_ddl
+  on ddl_command_start
```

Comment:
`ddl_command_start` is not compatible with `pg_event_trigger_ddl_commands()`

currently that function will always return `null`, if you change it to `ddl_command_end` it will work as you expect